ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -286,8 +286,12 @@ public static Schema resolveBinaryOperatorResultType(
       return Schema.OPTIONAL_STRING_SCHEMA;
     }
 
-    if (DecimalUtil.isDecimal(left) && DecimalUtil.isDecimal(right)) {
-      return resolveDecimalOperatorResultType(left, right, operator);
+    if (DecimalUtil.isDecimal(left) || DecimalUtil.isDecimal(right)) {
+      if (left.type() != Schema.Type.FLOAT64 && right.type() != Schema.Type.FLOAT64) {
+        return resolveDecimalOperatorResultType(
+            DecimalUtil.toDecimal(left), DecimalUtil.toDecimal(right), operator);
+      }
+      return Schema.OPTIONAL_FLOAT64_SCHEMA;
```

Comment:
Would this lose the precision if a math operation with decimals is converted to a float64? Don't we want to keep the precision of the resulted decimal?

Reply:
We do lose it - see above, this is desired.